common/lib/xmodule/xmodule/modulestore/tests/test_publish.py

```diff
@@ -139,3 +157,1222 @@ def test_publish_draft_delete(self):
+            """
+            self.assertEqual(xml_root.tag, tag)
+            if attrs:
+                for attr_name, attr_re in attrs.iteritems():
+                    if attr_re:
+                        self.assertIn(attr_name, xml_root.attrib)
+                        attr_re_comp = re.compile(attr_re)
+                        self.assertIsNotNone(
+                            attr_re_comp.search(xml_root.attrib[attr_name]),
+                            msg='Attr {} of tag {} doesn\'t match regex of:\n{}'.format(
+                                attr_name, tag, attr_re
+                            )
+                        )
+            else:
+                # If not checking the attrs, there should be *no* attrs in the XML.
+                self.assertEqual(xml_root.attrib, {})
+            if children:
+                for child in xml_root:
+                    _assert_xml_level(child, children['tag'], children.get('attrs'), children.get('children'))
+            else:
+                # If not checking the children, there should be *no* children in the XML.
+                self.assertEqual([child for child in xml_root], [])
+
+        # Parse the XML string into an ElementTree.
+        block_tree = ET.fromstring(block_contents)
+        # Now go through and verify the individual tags/attributes/children.
+        _assert_xml_level(block_tree, checklist['tag'], checklist.get('attrs'), checklist.get('children'))
+
+    def assertOLXContent(self, block_type, block_id, **kwargs):
+        """
+        Assert that a particular block exists in a course export in the proper draft/published location
+        and that the format is correct.
+        """
+        course_export_dir = self._get_course_export_dir()
+        is_draft = kwargs.pop('draft', False)
+        xml_to_check = kwargs.pop('xml', None)
+        xml_re_to_check = kwargs.pop('xml_re', None)
+        xml_parse = kwargs.pop('xml_parse', None)
+
+        block_path = self._get_block_type_path(course_export_dir, block_type, is_draft)
+        block_contents = self._get_block_contents(block_path, block_id)
+        if xml_to_check:
+            self.assertIn(xml_to_check, block_contents)
+        if xml_re_to_check:
+            xml_re = re.compile(xml_re_to_check)
+            self.assertIsNotNone(
+                xml_re.search(block_contents),
+                msg='Block ({}, {}) contents of:\n{}\n don\'t match regex of:\n{}'.format(
+                    block_type, block_id, block_contents, xml_re_to_check
+                )
+            )
+        if xml_parse:
+            self._assert_parsed_xml(block_contents, xml_parse)
+
+    def assertOLXMissing(self, block_type, block_id, **kwargs):
+        """
+        Assert that a particular block does not exist in a particular draft/published location.
+        """
+        course_export_dir = self._get_course_export_dir()
+        is_draft = kwargs.pop('draft', False)
```

Comment:
Good use of a keyword only argument! It would help to document it in the docstring, though.